Ginger/Ginger/ExternalConfigurations/GingerAnalyticsConfigurationPage.xaml.cs

```diff
@@ -0,0 +1,234 @@
+            if (AreRequiredFieldsEmpty())
+            {
+                Reporter.ToUser(eUserMsgKey.RequiredFieldsEmpty);
+                return;
+            }
+
+            GingerCoreNET.GeneralLib.General.CreateGingerAnalyticsConfiguration(gingerAnalyticsUserConfig);
+
+            if (IsTokenValid())
+            {
+                Reporter.ToUser(eUserMsgKey.GingerAnalyticsConnectionSuccess);
+                return;
+            }
+
+            bool isAuthorized = await HandleTokenAuthorization();
+            ShowConnectionResult(isAuthorized);
+        }
+
+        private bool AreRequiredFieldsEmpty()
+        {
+            return string.IsNullOrEmpty(gingerAnalyticsUserConfig.AccountUrl)
+                || string.IsNullOrEmpty(gingerAnalyticsUserConfig.IdentityServiceURL)
+                || string.IsNullOrEmpty(gingerAnalyticsUserConfig.ClientId)
+                || string.IsNullOrEmpty(gingerAnalyticsUserConfig.ClientSecret);
+        }
+
+        private async Task<bool> HandleTokenAuthorization()
+        {
+            if (string.IsNullOrEmpty(gingerAnalyticsUserConfig.Token))
+            {
+                return await RequestToken(gingerAnalyticsUserConfig.ClientId, gingerAnalyticsUserConfig.ClientSecret, gingerAnalyticsUserConfig.IdentityServiceURL);
+            }
+
+            return true;
+        }
+
+        private static void ShowConnectionResult(bool isAuthorized)
+        {
+            if (isAuthorized)
+            {
+                Reporter.ToUser(eUserMsgKey.GingerAnalyticsConnectionSuccess);
+            }
+            else
+            {
+                Reporter.ToUser(eUserMsgKey.GingerAnalyticsConnectionFail);
+            }
+        }
+
+
+        private void xClientSecretTextBox_LostKeyboardFocus(object sender, System.Windows.Input.KeyboardFocusChangedEventArgs e)
+        {
+            if (!EncryptionHandler.IsStringEncrypted(xClientSecretTextBox.ValueTextBox.Text))
+            {
+                xClientSecretTextBox.ValueTextBox.Text = ValueExpression.IsThisAValueExpression(xClientSecretTextBox.ValueTextBox.Text) ? xClientSecretTextBox.ValueTextBox.Text : EncryptionHandler.EncryptwithKey(xClientSecretTextBox.ValueTextBox.Text);
+
+            }
+        }
+
+        private void xClientIdTextBox_LostKeyboardFocus(object sender, System.Windows.Input.KeyboardFocusChangedEventArgs e)
+        {
+            if (!EncryptionHandler.IsStringEncrypted(xClientIdTextBox.ValueTextBox.Text))
+            {
+                xClientIdTextBox.ValueTextBox.Text = ValueExpression.IsThisAValueExpression(xClientIdTextBox.ValueTextBox.Text) ? xClientIdTextBox.ValueTextBox.Text : EncryptionHandler.EncryptwithKey(xClientIdTextBox.ValueTextBox.Text);
+
+            }
+        }
+
+
+        public async Task<bool> RequestToken(string clientId, string clientSecret, string address)
+        {
+            try
+            {
+                HttpClientHandler handler = new HttpClientHandler() { UseProxy = false};
```

Comment:
Create a separate class to do interaction with analytics API so that it can be easily mocked and unit tested